learning/dashboard/WHAT IS JAVASCRIPT/Basics of JavaScript.md

```diff
@@ -0,0 +1,941 @@
+   console.log(Math.round(1.49)); // Output: 1
+   console.log(Math.round(1.5)); // Output: 2
+
+**In the example above, Math.round() rounds 1.49 down to 1 and 1.5 up to 2.**
+
+[*Note that Math.round() returns a new rounded integer, it does not modify the original number*]
+
+<br>
+
+## How to Traverse the DOM Using JavaScript
+
+DOM traversing describes how to navigate the tree-like structure that
+HTML documents generate. Here's a complete guide on how to traverse the
+DOM with JavaScript.
+
+### What Is DOM Traversing?
+
+The Document Object Model, or DOM for short, is a tree-like
+representation of an HTML document. It pro vides an API that allows you,
+as the web developer, to interact with a website using JavaScript.
+
+Every item in the DOM is known as a node. Only through the DOM can you
+manipulate your HTML docu ment structure, content, and style.
+
+DOM traversal (also called walking or navigating the DOM) is the act of
+selecting nodes in the DOM tree f rom other nodes. You're probably
+already familiar with several methods for accessing elements in the DO M
+tree by their id, class, or tag name. You can use methods like
+document.querySelector() and document .getElementById() to do so.
+
+There are other methods you can use in conjunction, to navigate the DOM
+in more efficient and robust wa ys. As you can imagine, it's better to
+search from an already-known point on a map than do a full search.
+
+For example, selecting a child element from its parent is easier and
+more efficient than searching for it thr oughout the entire tree.
+
+## A Sample Document to Traverse
+Once you have access to a given node in the DOM tree, you can access its
+related nodes in different way s. You can move downwards, upwards, or
+sideways in the DOM tree from your selected node.
+
+The first method involves searching for an element starting with a
+top-most node (like the document node) and moving downwards.
+
+The second way is the opposite: you move from an inner element up the
+tree, searching for an outer elem ent. The last method is when you
+search for an element from another element at the same level (meaning
+the two elements are siblings) in the document tree.
+
+### To demonstrate, consider this example HTML document:
+
+
+Walking the DOM is a handy technique you can use in your web apps with
+just a little practice.
+
+### A large JavaScript logo on a blue background
+Readers like you help support MUO. When you make a purchase using links
+on our site, we may earn an affiliate commission. Read More.
+
+Understanding the DOM is essential in your web development career. You
+should know how to select diff erent elements in the DOM, so you can
+then read their contents or modify them.
+
+DOM traversing describes how to navigate the tree-like structure that
+HTML documents generate. Here's a complete guide on how to traverse the
+DOM with JavaScript.
+
+### What Is DOM Traversing?
```

Comment:
Looks like the content is repeated. Please check the entire document again and remove any duplicate sections.